cpp/src/interop/to_arrow.cu

```diff
@@ -220,6 +220,7 @@ std::shared_ptr<arrow::Array> dispatch_to_arrow::operator()<numeric::decimal128>
   rmm::cuda_stream_view stream)
 {
   using DeviceType = __int128_t;
+  auto constexpr max_precision = 38;
```

Comment:
Hard coding the number of max precision is a bit confusing because somebody may not know at all "where is the number (18) 38 coming from?". 

It makes more sense to formulate it, like:
```
template<typename T>
constexpr std::size_t max_precision() {
  auto const num_bits = sizeof(T) * 8;
  return std::floor(num_bits * std::log(2) / std::log(10));
}
```

Reply:
Good idea. Added a util function as suggested and replaced the hard-coded max precision with calling the util function.